stdlib/public/Concurrency/Task.swift

```diff
@@ -510,22 +507,17 @@ extension Task {
   /// asynchronous function present in this functions call stack.
   ///
   /// The returned value must not be accessed from tasks other than the current one.
-  @available(*, deprecated, message: "Not implemented yet, use Builtin.getCurrentAsyncTask() or Task.___unsafeCurrentAsync() until this function is implemented.")
   public static var unsafeCurrent: UnsafeCurrentTask? {
-    // FIXME: rdar://70546948 implement this once getCurrentAsyncTask can be called from sync funcs
-    //    guard let _task = Builtin.getCurrentAsyncTask() else {
-    //      return nil
-    //    }
-    //    return UnsafeCurrentTask(_task)
-    fatalError("\(#function) is not implemented yet")
+    guard let _task = _getActiveAsyncTask() else {
+      return nil
+    }
+    // FIXME: This retain seems pretty wrong, however if we don't we WILL crash
+    //        with "destroying a task that never completed" in the task's destroy.
+    //        How do we solve this properly?
+    _swiftRetain(_task)
```

Comment:
How should we actually solve this? See the issue explained in

Reply:
Hmm. I don't think we can express unowned return values on the Swift side, so the retain is correct.

Reply:
I see, thank you. We could put it into the C function to be less ugly, but it felt more normal to retain as close to where we "have to" as possible, so on this outer layer